test/e2e/network/network_tiers.go

```diff
@@ -0,0 +1,165 @@
+	})
+	It("should be able to create and tear down a standard-tier load balancer [Slow]", func() {
+		lagTimeout := framework.LoadBalancerLagTimeoutDefault
+		createTimeout := framework.GetServiceLoadBalancerCreationTimeout(cs)
+
+		svcName := "net-tiers-svc"
+		ns := f.Namespace.Name
+		jig := framework.NewServiceTestJig(cs, svcName)
+
+		By("creating a pod to be part of the service " + svcName)
+		jig.RunOrFail(ns, nil)
+
+		By("creating a Service of type LoadBalancer using the standard network tier")
+		svc := jig.CreateTCPServiceOrFail(ns, func(svc *v1.Service) {
+			svc.Spec.Type = v1.ServiceTypeLoadBalancer
+			setNetworkTier(svc, gcecloud.NetworkTierAnnotationStandard)
+		})
+		// Record the LB name for test cleanup.
+		serviceLBNames = append(serviceLBNames, cloudprovider.GetLoadBalancerName(svc))
+
+		svc = jig.WaitForLoadBalancerOrFail(ns, svcName, createTimeout)
+		lbIngress := &svc.Status.LoadBalancer.Ingress[0]
+		svcPort := int(svc.Spec.Ports[0].Port)
+		ingressIP := framework.GetIngressPoint(lbIngress)
+
+		By("running sanity and reachability checks")
+		jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
+		jig.TestReachableHTTP(ingressIP, svcPort, lagTimeout)
+		// Check the network tier of the forwarding rule.
+		netTier, err := getLBNetworkTierByIP(ingressIP)
+		Expect(err).NotTo(HaveOccurred(), "failed to get the network tier of the load balancer")
+		Expect(netTier).To(Equal(gcecloud.NetworkTierStandard))
+
+		By("updating the Service to use the premium (default) tier")
+		existingIP := ingressIP
+		svc = jig.UpdateServiceOrFail(ns, svcName, func(svc *v1.Service) {
+			clearNetworkTier(svc)
+		})
+		// Wait until the ingress IP changes. Each tier has its own pool of
+		// IPs, so changing tiers implies changing IPs.
+		svc = jig.WaitForNewIngressIPOrFail(ns, svcName, existingIP, createTimeout)
+		lbIngress = &svc.Status.LoadBalancer.Ingress[0]
+		ingressIP = framework.GetIngressPoint(lbIngress)
+
+		By("running sanity and reachability checks")
+		jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
+		jig.TestReachableHTTP(ingressIP, svcPort, lagTimeout)
+		// Check the network tier of the forwarding rule.
+		netTier, err = getLBNetworkTierByIP(ingressIP)
+		Expect(err).NotTo(HaveOccurred(), "failed to get the network tier of the load balancer")
+		Expect(netTier).To(Equal(gcecloud.NetworkTierPremium))
+
+		// TODO: Add tests for user-requested IPs.
+	})
+})
+
+func getLBNetworkTierByIP(ip string) (gcecloud.NetworkTier, error) {
+	var rule *computealpha.ForwardingRule
+	// Retry a few times to tolerate flakes.
+	err := wait.PollImmediate(5*time.Second, 15*time.Second, func() (bool, error) {
+		obj, err := getGCEForwardingRuleByIP(ip)
+		if err != nil {
+			return false, err
+		}
+		rule = obj
+		return true, nil
+	})
+	if err != nil {
+		return "", err
+	}
+	return gcecloud.NetworkTierGCEValueToType(rule.NetworkTier), nil
+}
+
+func getGCEForwardingRuleByIP(ip string) (*computealpha.ForwardingRule, error) {
```

Comment:
Note for the reviewers: This test identifies the forwarding rule used by the LB by IP. Alternatively, I can use the `GetLoadBalancerName()` to find the forwarding rule with the given name. I went with the former because it seemed to be more generic and less implementation dependent, but I could be convinced to just go with the later.